kong/tools/responses.lua

```diff
@@ -123,7 +123,18 @@ local function send_response(status_code)
     end
 
     ngx.status = status_code
-    ngx.header["Server"] = server_header
+
+    if singletons and singletons.configuration then
+      if singletons.configuration.enabled_headers[constants.HEADERS.SERVER] then
+        ngx.header["Server"] = server_header
+
+      else
+        ngx.header["Server"] = nil
+      end
+
+    else
+      ngx.header["Server"] = server_header
+    end
```

Comment:
Ditto: probably belong to a separate commit (i.e. in the PR already opened for delayed response)

Reply:
I think you wanted to comment about request-termination plugin changes here? I will make separate PR for the two plugins.